### ISSUE TITLE
SentryCoreDataSwizzling: malloc: Heap corruption detected, free list is damaged at 0x28294e580 

### ISSUE THREAD

**Issue by AndrewSB** [getsentry/sentry-cocoa#3142]
### Platform

iOS

### Installed

Swift Package Manager

### Version

8.7.3

### Steps to Reproduce

not sure

### Expected Result

no crash

### Actual Result

[Crash#366_v1.0 (1).txt](https://github.com/getsentry/sentry-cocoa/files/12030398/Crash.366_v1.0.1.txt)

getting this crash, not sure if its a memory pressure crash, or it's legit. message is `malloc: Heap corruption detected, free list is damaged at 0x28294e580 *** Incorrect guard value: 36028797018964027`

**Comment by AndrewSB** [comment]
thread 11 is the crashing thread, apparently

**Comment by philipphofmann** [comment]
Thanks for reporting this @AndrewSB. Do you have performance monitoring enabled? 

We also see this happening for other users. Internal [SDK crashes link](https://sentry.sentry.io/issues/4299765761/?project=4505469596663808&query=is%3Aunresolved&referrer=issue-stream&statsPeriod=14d&stream_index=4).

**Comment by AndrewSB** [comment]
if that's the default, then yes. the only change we make to the options at start is `options.enableUserInteractionTracing = false`

**Comment by dfed** [comment]
I just tried updating from v0.37.0 to v0.38.0, and I'm seeing this crash a _ton_. > 10% of sessions. Reverting fixed. Let me know if I can help debug!

**Comment by philipphofmann** [comment]
> I just tried updating from v0.37.0 to v0.38.0, 

@dfed, are you referring to 8.37.0 and 8.38.0, cause we don't have the above mentioned versions for the Cocoa SDK.

**Comment by dfed** [comment]
@philipphofmann yes! 🤦‍♂️. Apologies. Thank you for clarifying. Editing the post above now.

**Comment by philipphofmann** [comment]
@dfed, thanks for updating the versions. Did you see the crash also in `v8.37.0`? 

> Let me know if I can help debug!

Yes, you could share stacktraces of such crashes. If you can reliably reproduce it, it would be amazing sharing a small sample application with us. If that doesn't work and you can reliably reproduce it in debug, the Sentry SDK debug logs, which you can enable by setting `options.debug = true` would be very helpful.

**Comment by dfed** [comment]
I tried 8.38.0 and 8.41.0, and both reproduced.

Stack traces were all over the place. I don't think I saw a crash in the same place twice. I'm on vacation + away from my work computer until Jan 5, but here are two stack traces from an internal beta build:

1. https://portola.sentry.io/issues/6160330396/events/a6c58b50304c4cbcb4788addde151d50/?project=4506735735865344
2. https://portola.sentry.io/issues/6160332466/events/ac993b23a233446a926dab3ab8fea1da/?project=4506735735865344

I don't think I'm going to be able to get you a small sample app since I don't have a clue what is causing this. My app has Unity embedded in it, and we have a dozen+ 3rd party libraries, including one that's using CoreData. Happy to send you a list though. And when I'm back at a computer I'd be happy to get you debug logs.

**Comment by philipphofmann** [comment]
Thanks, @dfed, for the extra information. That should be enough for us to investigate. I can't give you an ETA for this, though.

**Comment by dfed** [comment]
Glad those issue links were helpful! Here are debug logs from 8.43.0 between launch and the heap corruption crash. Heap corruption occurred just after launch:
[Sentry debug 8.43.0.log](https://github.com/user-attachments/files/18321746/Sentry.debug.8.43.0.log)
[Sentry debug 8.43.0 2.log](https://github.com/user-attachments/files/18321758/Sentry.debug.8.43.0.2.log)
[Sentry debug 8.43.0 3.log](https://github.com/user-attachments/files/18321763/Sentry.debug.8.43.0.3.log)

Gathering the above logs from three separate crashes took me less than 5min – this is very easy for me to reproduce on versions greater than 8.37.0. Happy to send more info if at all useful. You can also reach me on the iOS Folks slack if that'd be faster for iterating.

**Comment by dfed** [comment]
Had a hunch, and it looks like the culprit 3rd party dependency is https://github.com/intercom/intercom-ios-sp. If I remove that integration entirely on Sentry 8.43.0 I can't reproduce the heap corruption problem.

**Comment by philipphofmann** [comment]
@dfed, that's fantastic news thank you, because it either means we aren't the culprit or only when using both SDKs which helps us pinpoint the problem. Does the integration also cause problems without Sentry?

**Comment by dfed** [comment]
It's a combination thing as far as I can tell – I can't reproduce this issue without bumping Sentry. I know that Intercom also swizzles, and that they set up swizzling in `+load`, so they likely are swizzling before you are.

I spent an hour trying to create a sample app that reproduced the crash, but haven't been able to trigger it thus far. I wonder if there's another piece of the puzzle I haven't found yet. If I follow the hunch that swizzling is a likely culprit, CustomerIO might be involved... I'll keep poking.

**Comment by SleepiestAdam** [comment]
Can confirm we also see this issue when using /intercom-ios-sp. We don't use CustomerIO, so I don't think that's related. We do use Realm.

**Comment by SleepiestAdam** [comment]
Just to follow up, it also doesn't occur (at least for us) if you don't enable profiling. If you do enable SentrySDK.startProfiler then the crash occurs.

**Comment by CraigSiemens** [comment]
We recently ran into this crash as well after updating from 8.36.0 to 8.43.0. We then updated to 8.44.0 and kept seeing crashes. For now we've downgraded to 8.37.0 and the crashes have stopped.

<details><summary>Here's some screenshots of the crash stack traces.</summary>
<p>

![Image](https://github.com/user-attachments/assets/29919a6e-afd5-40fe-b6f9-4a56aa1cd5b6)
![Image](https://github.com/user-attachments/assets/1928d865-6ec4-41db-b0b5-47822656f89c)
![Image](https://github.com/user-attachments/assets/36f23cc2-2803-4ff1-a6b1-61252dc6aa9f)
![Image](https://github.com/user-attachments/assets/e4ecb7d2-5090-4ac9-9da8-4fda90bb7510)

</p>
</details>

**Comment by philprime** [comment]
Thanks @CraigSiemens for the info. I believe the reason for the crashes you are seeing are related to #4887

**Comment by filipef101** [comment]
We're seeing this too, on ios simulator it's constant. on real devices it's a bit random, but the devices/users that get it, get it often.
Caused by enabling swizzling.

sentry cocoa v8.45.0


With this config there doesn't seem to be any error:
      options.enableUIViewControllerTracing = YES;
      options.enableSwizzling = NO;
      options.enablePreWarmedAppStartTracing = YES;
      options.attachViewHierarchy = NO;

With this config there is:
      options.enableUIViewControllerTracing = YES;
      options.enableSwizzling = YES;
      options.enablePreWarmedAppStartTracing = YES;
      options.attachViewHierarchy = NO;

**Comment by kahest** [comment]
@filipef101 thanks for writing in - we shipped a fix for a potentially related issue in [8.46.0](https://github.com/getsentry/sentry-cocoa/blob/main/CHANGELOG.md#8460), could you try if upgrading improves the situation for you?

**Comment by filipef101** [comment]
> @filipef101 thanks for writing in - we shipped a fix for a potentially related issue in [8.46.0](https://github.com/getsentry/sentry-cocoa/blob/main/CHANGELOG.md#8460), could you try if upgrading improves the situation for you?

Has react native package been updated to that version?

**Comment by philprime** [comment]
@filipef101 looking at the [releases](https://github.com/getsentry/sentry-react-native/releases) it seems that it still uses 8.45.0 as updated in v6.8.0.

**Comment by denrase** [comment]
Unfortunately, i was not able to reproduce this.

- Sentry 8.45.0
- Intercom 19.4.1

```swift
options.tracesSampleRate = 1.0
options.enableAutoPerformanceTracing = true
options.profilesSampleRate = 1.0
options.profilesSampler = { _ in return true }
options.enableUIViewControllerTracing = true
options.enableSwizzling = true
options.enablePreWarmedAppStartTracing = true
options.attachViewHierarchy = false
```

Created a stress test with hundreds core data and network calls, of which there were also sentry spans created through swizzling, while also interacting with intercom at the same time, but could not produce any crash.

@filipef101 @CraigSiemens @dfed @SleepiestAdam Is any of you still seeing these issues and could you provide us with a more recent logs or even a repro? Thanks! 🙇

**Comment by dfed** [comment]
I turned off profiling a year ago and haven't tried turning it on since. Apologies but I don't think I can spend more time to investigate here.

**Comment by philippsuess** [comment]
Thanks for the update, @dfed. That's understandable.

**Comment by denrase** [comment]
@dfed Thank you, in any case i will notify you when we find a solution here.

**Comment by dfed** [comment]
Just took a look at Intercom's latest SDK and they no longer include Sentry symbols in their binary. So this issue has likely been resolved at the root! Going to try turning this back on. Will report if I run into issues.

**Comment by denrase** [comment]
@dfed Did you experience this issue? I'm closing the issue for now, but can re-open if this pops up again.

**Comment by dfed** [comment]
@denrase thanks for the follow up. I have not experienced this after rolling out. Closing SGTM
